velox/common/file/File.cpp

```diff
@@ -377,6 +377,21 @@ void LocalWriteFile::write(
 void LocalWriteFile::truncate(int64_t newSize) {
   checkNotClosed(closed_);
   VELOX_CHECK_GE(newSize, 0, "New size cannot be negative.");
+#ifdef linux
+  if (newSize > size_) {
+    // Use fallocate to extend the file.
+    const auto ret = ::fallocate(fd_, 0, 0, newSize);
```

Comment:
So we will add fallocate support in followup by using setAttr? thanks!

Reply:
What will this attr do? Meaning if set, we use fallocate instead of truncate?

Reply:
This is not required.